dashboard/test/lib/services/script_seed_test.rb

```diff
@@ -1108,6 +1108,7 @@ def create_script_tree(
         )
         CourseOffering.add_course_offering(script)
       end
+      script.reload
```

Comment:
I don't know why I needed to do this in so many tests. Anyone have any ideas?

Reply:
I'm not sure where this is documented, but my general understanding is like this:
1. ActiveRecord objects like `script` here cache what they know about the state of the DB
2. if you use the object to update the contents of the DB as with `script.update!`, the cached data in the model is updated (in addition to the DB contents).
3. however, in this case, `CourseOffering.add_course_offering` calls `CourseVersion.add_course_version`, which updates an associated object of the `script` object without actually touching the `script` object: https://github.com/code-dot-org/code-dot-org/blob/32172ea9a808720074abe6d54d74adaaf5c7b352/dashboard/app/models/course_version.rb#L73 therefore, the `script` object has stale cached contents, and still thinks it does not have a course_version.
4. if instead `CourseVersion.add_course_version` looked like this, you might not need to reload after:
```ruby
  def self.add_course_version(course_offering, content_root)
    if content_root.is_course?
      content_root.course_version = ...
```

Reply:
it is also possible you could keep the `add_course_version` method how it is and add a `content_root.reload_course_version` call to the end of it, which I just read about here: https://guides.rubyonrails.org/v5.2/association_basics.html#methods-added-by-has-one-association

Reply:
That makes sense as it seems to be happening in other tests as a result of `create :unit_group_unit` which would also not update the active record cache. Curious why these two methods are causing this in so many cases such as this one.